src/main.py

```diff
@@ -1264,9 +1253,11 @@ def _return_to_cluster(self, new_primary, role, is_dead=False):
             last_op = self.zk.noexcept_get('%s/%s/op' % (self.zk.MEMBERS_PATH, helpers.get_hostname()))
             logging.info('Last op is: %s' % str(last_op))
             if role != 'primary' and not self.is_op_destructive(last_op) and not self._is_simple_primary_switch_tried():
-                logging.info('Trying to do a simple primary switch.')
+                logging.info('Trying to do a simple WAL source switch to: {}'.format(new_primary))
                 result = self._try_simple_primary_switch_with_lock(limit, new_primary, is_dead)
-                logging.info('Primary switch count: %s finish with result: %s', self.checks['primary_switch'], result)
+                logging.info('WAL source switch count: %s finish with result: %s', self.checks['primary_switch'], result)
```

Comment:
I think, naming a variable `primary_switch`, and logging it as `WAL source switch` will make debugging more obscure.